### ISSUE TITLE
Flamegraph: Better zoom focus interaction

### ISSUE THREAD

**Issue by cyriltovena** [grafana/grafana#58373]
When investigating flamegraph, if one node is too big, the rest is too big to grab. It would be nice if we could have a way to hide big nodes via right+click or using a regex.

See the example below:

![image](https://user-images.githubusercontent.com/1053421/191921350-44506c2c-e9bc-4bfb-9d25-4153a1052083.png)

**Comment by chancez** [comment]
I have a similar, issue that this _might_ solve. I don't generally have "big" nodes, but I have a handful of nodes, where a lot of time is spent/go routines are, and so some nodes are just too small to see (we have a lot of grey nodes like yours does on the edges). I think for my case, this would help, but the ability to "zoom" to a particular section of the graph might help. Would that be worth considering as well?

Eg:
![Screenshot 2022-12-15 at 3 47 44 PM](https://user-images.githubusercontent.com/1400401/207999874-1669333e-7d4e-4ee5-a895-450af3a3fd54.png)

**Comment by leeoniya** [comment]
we've discussed implementing the same kind of x-selection [brush] zoom as we have in the TimeSeries panel (and graph viz above the flamegraph). it would likely work better than going through and carefully hiding multiple specific sibling branches.

**Comment by chancez** [comment]
> (feels kinda backwards to have to interact with branches you're not interested in).

Right, that expresses what I was thinking perfectly. Is there an issue for this already?

**Comment by leeoniya** [comment]
> Is there an issue for this already?

in my mind this issue is it, since imo the goal is one and the same: zoom to unclickable/merged regions. i'm not sure we'll want to implement _both_ zooming and ignoring of branches unless there's significant value to having/maintaining both.

**Comment by chancez** [comment]
👍 seems reasonable. Just wanted to make sure this was tracked somehow.

**Comment by leeoniya** [comment]
cool! but i'm not sure this MVP will be applicable to the direct flamegraph renderer that will be replacing the current one in the next few months.

**Comment by chancez** [comment]
> cool! but i'm not sure this MVP will be applicable to the direct flamegraph renderer that will be replacing the current one in the next few months.

No worries. Should I skip creating a PR then?

**Comment by leeoniya** [comment]
i don't have an exact ETA for the migration, so this could well be useful before then. might as well PR it :)
